lib/repository/club_member/club_member_repository.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:dio/dio.dart';
+import 'package:woohakdong/model/club_member/club_member.dart';
+import 'package:woohakdong/service/dio/dio_service.dart';
+import 'package:woohakdong/service/logger/logger.dart';
+
+class ClubMemberRepository {
+  final Dio _dio = DioService().dio;
+
+  Future<List<ClubMember>> getClubMemberList(int clubId, DateTime? clubMemberAssignedTerm) async {
+    try {
+      logger.i('동아리 회원 목록 조회 시도');
+
+      final response = await _dio.get(
+        '/club/$clubId/members',
+        queryParameters: clubMemberAssignedTerm != null
+            ? {'clubMemberAssignedTerm': clubMemberAssignedTerm.toIso8601String()}
+            : {},
+      );
+
+      if (response.statusCode == 200) {
+        final Map<String, dynamic> jsonData = response.data;
+
+        List<dynamic> clubMemberListData = jsonData['result'] as List<dynamic>;
+
+        return clubMemberListData
+            .map((json) => ClubMember.fromJson(json as Map<String, dynamic>))
+            .toList();
+      } else {
+        throw Exception('동아리 회원 목록 조회 실패: 상태 코드 ${response.statusCode}');
+      }
```

Comment:
에어비앤비 javascript 컨벤션은 if문에서 return하면 뒤에 else는 불필요하다는 이유로 안 붙이는 것을 권장하는데 [16.3](https://github.com/ParkSB/javascript-style-guide#blocks--no-else-return), dart에선 컨벤션이 다른 건지 궁금하네요!

```suggestion
      if (response.statusCode == 200) {
        final Map<String, dynamic> jsonData = response.data;

        List<dynamic> clubMemberListData = jsonData['result'] as List<dynamic>;

        return clubMemberListData
            .map((json) => ClubMember.fromJson(json as Map<String, dynamic>))
            .toList();
      }
      
      throw Exception('동아리 회원 목록 조회 실패: 상태 코드 ${response.statusCode}');
```

Reply:
앗..자바스크립트 컨벤션은 그렇군요!

그냥 이건 다트 컨벤션 안보고 if 뒤에 else를 제가 그냥 붙였습니다...하하하ㅏ 🙄

Reply:
생각해보니 return으로 하면 else를 안붙이는게 더 좋을 것 같네요! 감사합니다 ㅎㅎㅎ

Reply:
아하 그렇군요ㅎㅎ